### PR TITLE
Added CCB commands (add, rm, approve, block) and associated changes t…

### PR DESCRIPTION
**Pull request by dawzee**
…o store in ticket, show status, read ccb members from config and prevent ticket transition without approval

### CODE REVIEW

**Review by aleb — CHANGES_REQUESTED** [review]
some nitpicks

File: commands/ccb_rm.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	_select "github.com/daedaleanai/git-ticket/commands/select"
+	"github.com/daedaleanai/git-ticket/entity"
+	"github.com/spf13/cobra"
+)
+
+func newCcbRmCommand() *cobra.Command {
+	env := newEnv()
+
+	cmd := &cobra.Command{
+		Use:      "rm <user> [<id>]",
```
[the remark this answers is not in the document]
Comment (dawzee) [reply commands/ccb_rm.go:18]: Hmm, should be optional as long as you have a ticket selected:
```
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket show --field ccb
Ian Whittington (Approved)
Martin Beeby (Added)
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket ccb rm Martin
Removing Martin Beeby from CCB group of ticket 6873dbb
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket show --field ccb
Ian Whittington (Approved)
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket ccb add Martin
Adding Martin Beeby to CCB group of ticket 6873dbb
```

with no ticket selected you get an error:
```
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket deselect
iw@iw:~/sandpit/git-ticket/one-id-and-bug(master)$ git ticket ccb rm Martin
Error: you must provide a ticket id or use the "select" command first
```

File: commands/ccb_rm.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		return fmt.Errorf("no users matching %s", userToRemoveString)
+	}
+
+	userToRemoveIdentity, err := env.backend.ResolveIdentity(userToRemoveId)
+	if err != nil {
+		return err
+	}
+
+	if b.Snapshot().GetCcbState(userToRemoveId) == bug.RemovedCcbState {
+		return errors.New(userToRemoveIdentity.DisplayName() + " is not in the ticket CCB group")
```
[the remark this answers is not in the document]
Comment (dawzee) [reply commands/ccb_rm.go:87]: good point, the ticket is already is already in the intended state so returning an error seems counter-intuitive. fixed here and in other commands.

File: bug/ccb.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		return "Blocked"
+	case RemovedCcbState:
+		return "Removed"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// ccbMembers holds a map of users who are in the master CCB member list, which is defined in the git ticket
+// configuration. Configurations must be of the form "map[string]interface{}" so is stored as {"<user id>":1}.
```
[the remark this answers is not in the document]
Comment (dawzee) [reply bug/ccb.go:49]: good idea. fixed.

File: commands/ccb_rm.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleb) [reply commands/ccb_rm.go:18]: nice! :)

**Review by aleb — APPROVED** [review]
LGTM

**Pull request merged**